### PR TITLE
MA-252: Limit Exceeded: Error while deploying all managed aws services in all aws regions

### PR DESCRIPTION
**Pull request by chandramishra**
Purpose:  Size of Trust Relationship Policy limit Exceeded to federate role in master account
‌
For more details Please refer this link : https://mountdiablo.jira.com/browse/MA-252?filter=-1
Changes:  API & particles ,lambda function
Integration testing/functional testing:
Require to test all managed-aws functionality
‌
Unit Testing
Not Required

### CODE REVIEW

File: awscfn-stack-launcher/index_cfn_launcher.js [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
-  if (event.federateAccount) {
-    roles.push({roleArn:'arn:aws:iam::' + event.federateAccount + ':role/' + event.federateRoleName});
-    var admin_role = {roleArn:'arn:aws:iam::' + event.account + ':role/' + event.roleName};
-    if (event.roleExternalId) {
-      admin_role.externalId = event.roleExternalId;
-    }
-    roles.push(admin_role);
-  }
-  console.log(roles);
+  //if (!event.federateRoleName)  event.federateRoleName = "federate";
```
Comment (SonalOjha) [inline awscfn-stack-launcher/index_cfn_launcher.js]: Remove the lines if not needed

File: awscfn-stack-launcher/index_cfn_launcher.js [hunk 2]
```diff
@@ -55,10 +63,11 @@ exports.handler = function (event, context) {
         input.bucketName = event.bucketName;
         input.policyDocument = JSON.stringify(policy);
         input.account = event.account;
+        input.selfAccount = true;  // this flag used for check whether code require execute in master or customer account
      
         var flows = [
-           {func:aws_s3.updatePolicy, success:aws_sts.assumeRoles, failure:failed, error:errored},
-           {func:aws_sts.assumeRoles, success:aws_cfn.createCfnStack, failure:failed, error:errored},
+           {func:aws_s3.updatePolicy, success:aws_cfn.createCfnStack, failure:failed, error:errored},
+           //{func:aws_sts.assumeRoles, success:aws_cfn.createCfnStack, failure:failed, error:errored},
```
Comment (SonalOjha) [inline awscfn-stack-launcher/index_cfn_launcher.js]: line would not be needed

File: lib/aws/lambda.js [hunk 3]
```diff
@@ -158,9 +158,56 @@ function AWSLambda() {
...
+	console.log("added permission");
+    }
+
     var lambda = me.preRun(self, input);
     lambda.addPermission(params, me.callback);
   }
+
+   me.assumeRoles = function(input, callback) {
+    console.log("=====started calling lambda function for get credential")
+    input.authorizer_user_guid="b66fdc22-edaa-4710-9ff9-6a77ec834179";
```
Comment (SonalOjha) [inline lib/aws/lambda.js]: What guid is this?

File: awscfn-stack-launcher/index_cfn_launcher.js [hunk 4]
```diff
@@ -55,36 +50,30 @@ exports.handler = function (event, context) {
         input.bucketName = event.bucketName;
         input.policyDocument = JSON.stringify(policy);
         input.account = event.account;
+        input.selfAccount = true;  // this flag used for check whether code require execute in master or customer account
```
Comment (RajkumarShivage) [inline awscfn-stack-launcher/index_cfn_launcher.js]: What is the context for this?

File: lib/aws/lambda.js [hunk 5]
```diff
@@ -158,9 +158,56 @@ function AWSLambda() {
...
+        input.creds=input.custAccount;
+      }
+	console.log("added permission");
+    }
+
     var lambda = me.preRun(self, input);
     lambda.addPermission(params, me.callback);
   }
+
+   me.assumeRoles = function(input, callback) {
```
Comment (RajkumarShivage) [inline lib/aws/lambda.js]: can we use "assumeRolesByLambda" here?

File: lib/aws/lambda.js (hunk 3 above)
Comment (chandramishra) [reply lib/aws/lambda.js]: Right now this value require to auth lambda as parameter but authlambda is not using this value
so we can give any value

Comment (SonalOjha) [reply lib/aws/lambda.js]: If no utility cant we send it empty string and add a comment to it

**Comment by chandramishra** [comment]
I think it will work with empty value (authorizer_user_guid="")

**Pull request merged**